editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -5501,17 +5501,17 @@ bool CanvasItemEditorViewport::_create_instance(Node *parent, String &path, cons
 	editor_data->get_undo_redo().add_undo_method(ed, "live_debug_remove_node", NodePath(String(edited_scene->get_path_to(parent)) + "/" + new_name));
 
 	CanvasItem *parent_ci = Object::cast_to<CanvasItem>(parent);
+	Vector2 target_pos = canvas_item_editor->get_canvas_transform().affine_inverse().xform(p_point);
+	target_pos = canvas_item_editor->snap_point(target_pos);
 	if (parent_ci) {
-		Vector2 target_pos = canvas_item_editor->get_canvas_transform().affine_inverse().xform(p_point);
-		target_pos = canvas_item_editor->snap_point(target_pos);
 		target_pos = parent_ci->get_global_transform_with_canvas().affine_inverse().xform(target_pos);
-		// Preserve instance position of the original scene.
-		CanvasItem *instance_ci = Object::cast_to<CanvasItem>(instantiated_scene);
-		if (instance_ci) {
-			target_pos += instance_ci->_edit_get_position();
-		}
-		editor_data->get_undo_redo().add_do_method(instantiated_scene, "set_position", target_pos);
 	}
+	// Preserve instance position of the original scene.
+	CanvasItem *instance_ci = Object::cast_to<CanvasItem>(instantiated_scene);
+	if (instance_ci) {
+		target_pos += instance_ci->_edit_get_position();
+	}
+	editor_data->get_undo_redo().add_do_method(instantiated_scene, "set_position", target_pos);
```

Comment:
What happens if the instantiated scene is not a CanvasItem and doesn't have `set_position`?

Reply:
There is an error.
It was already here actually, if the parent was CanvasItem. I'll fix it.